swaylock/render.c

```diff
@@ -18,7 +18,15 @@ static void set_color_for_state(cairo_t *cairo, struct swaylock_state *state,
 	} else if (state->auth_state == AUTH_STATE_CLEAR) {
 		cairo_set_source_u32(cairo, colorset->cleared);
 	} else {
-		cairo_set_source_u32(cairo, colorset->input);
+		if (state->xkb.caps_lock && state->args.show_caps_lock_indicator) {
+			cairo_set_source_u32(cairo, colorset->caps_lock);
+		} else if (state->xkb.caps_lock && !state->args.show_caps_lock_indicator) {
+			state->args.colors.text.input = state->args.colors.text.caps_lock;
+			cairo_set_source_u32(cairo, colorset->input);
+			state->args.colors.text.input = state->args.colors.text.input;
```

Comment:
You have to store the value before changing it. Currently, this line is a nop.

Reply:
Ahh, I see. This was total nonsense. Sorry for that